ssl/statem/statem_srvr.c

```diff
@@ -114,6 +114,17 @@ int ossl_statem_server_read_transition(SSL *s, int mt)
                 return 1;
             }
         }
+        if (mt == SSL3_MT_CLIENT_KEY_EXCHANGE && s->s3->tmp.cert_request
```

Comment:
It's really hard to parse this decision tree. Do you think you can reorganize it so all the `mt == SSL3_MT_CLIENT_KEY_EXCHANGE`logic falls under one `if`-clause. Something like:

```
if (mt == SSL3_MT_CLIENT_KEY_EXCHANGE) {
    if (s->s3->tmp.cert_request) {
        if (s->version == SSL3_VERSION) {
            /* look at verify mode */
        } else {
            /* fail: client should always send a Certificate message. */
        }
    } else {
        /* No certificate requested and none received: ok. */
    }
}
```

I think this separates the SSLv3 special casing a little better, what do you think?